src/containers/quiz/quiz-question/Question.tsx

```diff
@@ -0,0 +1,175 @@
+
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import FormGroup from '@mui/material/FormGroup'
+import RadioGroup from '@mui/material/RadioGroup'
+import AppCard from '~/components/app-card/AppCard'
+import CheckIcon from '@mui/icons-material/Check'
+import CloseIcon from '@mui/icons-material/Close'
+import Answer from '~/containers/quiz/question-answer/Answer'
+import { SxProps } from '@mui/material/styles'
+
+import { determineQuestionType } from '~/components/question-editor/QuestionEditor.constants'
+import { isCorrectAnswer } from '~/utils/is-correct-answer'
+import { spliceSx } from '~/utils/helper-functions'
+import { styles } from '~/containers/quiz/quiz-question/Question.styles'
+
+import { Question } from '~/types'
+import { AnswerStatusEnum } from '~/containers/quiz/question-answer/Answer.types'
+
+interface QuizQuestionProps {
+  question: Question
+  index: number
+  value: string | null | string[]
+  showCorrectAnswers?: boolean
+  showPoints?: boolean
+  showAnswersCorrectness?: boolean
+  isEditable?: boolean
+  appCardWrapper?: boolean
+  sx?: SxProps
+  handleInputChange: ChangeEventHandler
+  handleNonInputValueChange: (value: string | string[]) => void
+}
+
+const QuizQuestion: FC<QuizQuestionProps> = ({
+  appCardWrapper,
+  sx,
+  index,
+  showPoints,
+  showCorrectAnswers,
+  showAnswersCorrectness,
+  value,
+  question,
+  isEditable,
+  handleInputChange,
+  handleNonInputValueChange
+}) => {
+  const { t } = useTranslation()
+
+  const { isMultipleChoice, isOpenAnswer } = determineQuestionType(
+    question.type
+  )
+
+  const ContainerComponent = appCardWrapper ? AppCard : Box
+
+  const isCorrect = isCorrectAnswer(question, value)
+
+  const iconStyles = styles.icon(
+    isCorrect ? AnswerStatusEnum.Correct : AnswerStatusEnum.Incorrect
+  )
+
+  const correctnessIcon =
+    showAnswersCorrectness &&
+    (isCorrect ? <CheckIcon sx={iconStyles} /> : <CloseIcon sx={iconStyles} />)
+
+  const shouldShowCorrectAnswers = showCorrectAnswers && !isOpenAnswer
+
+  const correctAnswersList =
+    shouldShowCorrectAnswers &&
+    question.answers
+      .filter((item) => item.isCorrect)
+      .map((item) => (
+        <Answer
+          checked
+          isCorrect={item.isCorrect}
+          isEditable={false}
+          key={item.text}
+          label={item.text}
+          showCorrectness
+          text={item.text}
+          type={question.type}
+        />
+      ))
+
+  const correctAnswers = shouldShowCorrectAnswers && (
+    <Box sx={styles.correctAnswers.root}>
+      <Typography sx={styles.correctAnswers.title}>
+        {t('myResourcesPage.quizzes.correctAnswers')}
+      </Typography>
+      <Box sx={styles.correctAnswers.list}>{correctAnswersList}</Box>
+    </Box>
+  )
+
+  const answersList = question.answers.map((answer) => {
+    const isChecked = Boolean(
+      isMultipleChoice ? value?.includes(answer.text) : value === answer.text
+    )
```

Comment:
I think wrapping in `Boolean` is redundant because both value.includes and === produce boolean output

Reply:
Since value can be null, without `Boolean` isChecked could be undefined what resulted in error in console (error about component that is uncontrolled initially and than becomes controlled). I changed it to the following, I guess it would be less confusing:
```
const isValidMultipleChoice = value && isMultipleChoice
const isChecked = isValidMultipleChoice
      ? value.includes(answer.text)
      : value === answer.text
```

Reply:
What do you think about refactoring just like following:

```jsx
value && isMultipleChoice ? value.includes(answer.text) : value === answer.text
```

Reply:
Thanks, fixed